### PR TITLE
#164069195 bookmark articles

### PR DESCRIPTION
**Pull request by Peace-Apple**
## What does this PR do?

This PR implements the feature of a user being able to bookmark or unbookmark an article.

## Description of the task to be completed

- Enable users to bookmark an article
- Enable users to unbookmark an article

## How should this be manually tested?

- Clone this repository
- cd into the directory Ah-aquaman-frontend
- Check out the branch `ft-bookmark-articles-164069195`
- Run `npm install` to install all dependencies
- Run`npm start` to get the server running
- Once you are logged in, you can view an article and bookmark it but you can not bookmark your own article
- You can also remove a bookmark from an article you had previously bookmarked

## What are the relevant pivotal tracker stories?

[#164069195](https://www.pivotaltracker.com/story/show/164069195)

## Screenshots

An article that has been bookmarked

image

An article that has been unbookmarked

image

### CODE REVIEW

File: src/views/articleDetailView/index.js [hunk 1]
```diff
@@ -52,18 +64,19 @@ export class ArticleDetailView extends Component {
     if (article.title === "") {
       return (
         <div className="h-100 w-100 p-5 m-5  text-center">
-          {toast.error("Article not found", {
+          {/* {toast.error("Article not found", {
```
Comment (BagzieGracious) [inline src/views/articleDetailView/index.js]: can you please remove this commented code?

**Review by BagzieGracious — CHANGES_REQUESTED** [review]
Thank you for the good work @Peace-Apple, but please implement this feedback

File: src/views/articleDetailView/index.js (hunk 1 above)
Comment (Peace-Apple) [reply src/views/articleDetailView/index.js]: It is removed @BagzieGracious

Comment (Peace-Apple) [reply src/views/articleDetailView/index.js]: I have fixed that @BagzieGracious

**Review by BagzieGracious — APPROVED** [review]

**Review by joelethan — APPROVED** [review]
Good job @CryceTruly

File: package.json [hunk 2]
```diff
@@ -82,6 +82,7 @@
         "react-tagsinput": "^3.19.0",
         "react-time-ago": "^5.0.4",
         "react-toastify": "^5.0.0",
+        "react-tooltip": "^3.10.0",
```
Comment (CryceTruly) [inline package.json:85]: Great work @Peace-Apple, how is this package helpful.

File: src/reducers/bookmarkReducer/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import {
+  BOOKMARK_SUCCESS, UNBOOKMARK_SUCCESS, BOOKMARK_LIST_FETCHED, BOOKMARK_FAIL, 
+} from "../../actions/types";
+
+const initialState = {
+  bookmarks: [],
+  isBookmarked: false,
+};
+
+const getIsBookmarked = (bookmarks, slug) => {
```
Comment (CryceTruly) [inline src/reducers/bookmarkReducer/index.js:10]: Great work @Peace-Apple, I love the way you use this filter function, Looks clean and readable

**Review by CryceTruly — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (Peace-Apple) [reply package.json:85]: That package enables us to include a text that will appear whenever a user hovers over that bookmark icon, in this case I display a text bookmark or unbookmark accordingly, you could check out the deployment to see how that works

File: src/reducers/bookmarkReducer/index.js (hunk 3 above)
Comment (Peace-Apple) [reply src/reducers/bookmarkReducer/index.js:10]: Thanks

File: src/__tests__/views/articleDetailView/__test__/articleDetailView.test.js [hunk 4]
```diff
@@ -48,17 +52,17 @@ describe("ArticleDetailView", () => {
 
   it("should call getSingleArticle after will mount ", () => {
```
Comment (AmosWels) [inline src/__tests__/views/articleDetailView/__test__/articleDetailView.test.js]: improve this `it` title

File: src/components/article/articleComponent/index.js [hunk 5]
```diff
@@ -35,13 +36,22 @@ const ArticleComponent = props => (
...
-                <p>
+                <span>
                   { parseDate(props.article.created_at) }
                   <span className="ml-2">
                     <i className="fa fa-clock" /> 
                     {props.article.read_time}
-                  </span> 
-                </p>
+                  </span>
+                  {props.loggedInUsername === null || props.loggedInUsername === props.articleAuthorUsername ? ""
```
Comment (AmosWels) [inline src/components/article/articleComponent/index.js]: please check this indentation

**Review by AmosWels — CHANGES_REQUESTED** [review]
Goodwork @Peace-Apple , please improve the description on your PR to indicate that a user shouldn't bookmark their own article as well. 

I noticed some eslint issues and indentation. Look at the reviews and resolve them

File: src/components/article/articleComponent/index.js (hunk 5 above)
Comment (Peace-Apple) [reply src/components/article/articleComponent/index.js]: Thanks, I have fixed this now

File: src/__tests__/views/articleDetailView/__test__/articleDetailView.test.js (hunk 4 above)
Comment (Peace-Apple) [reply src/__tests__/views/articleDetailView/__test__/articleDetailView.test.js]: I have fixed this as well

Comment (Peace-Apple) [reply src/__tests__/views/articleDetailView/__test__/articleDetailView.test.js]: @AmosWels I have now fixed them

**Review by AmosWels — APPROVED** [review]

**Pull request merged**